src/sentry/notifications/notifications/digest.py

```diff
@@ -0,0 +1,144 @@
+from sentry.digests.utilities import (
+    get_digest_metadata,
+    get_personalized_digests,
+    should_get_personalized_digests,
+)
+from sentry.notifications.notifications.base import BaseNotification
+from sentry.notifications.notify import notify
+from sentry.notifications.types import ActionTargetType
+from sentry.notifications.utils import get_integration_link, has_alert_integration
+from sentry.notifications.utils.digest import (
+    get_digest_subject,
+    send_as_alert_notification,
+    should_send_as_alert_notification,
+)
+from sentry.notifications.utils.participants import get_send_to
+from sentry.types.integrations import ExternalProviders
+
+if TYPE_CHECKING:
+    from sentry.models import Project, User
+
+
+logger = logging.getLogger(__name__)
+
+
+class DigestNotification(BaseNotification):
+    def __init__(
+        self,
+        project: "Project",
+        digest: Digest,
+        target_type: ActionTargetType,
+        target_identifier: Optional[int] = None,
+    ) -> None:
+        super().__init__(project)
+        self.digest = digest
+        self.target_type = target_type
+        self.target_identifier = target_identifier
+
+    def get_participants(self) -> Mapping[ExternalProviders, Set["User"]]:
+        data_by_provider = get_send_to(
+            project=self.project,
+            target_type=self.target_type,
+            target_identifier=self.target_identifier,
+        )
+        return {
+            provider: data
+            for provider, data in data_by_provider.items()
+            if provider in [ExternalProviders.EMAIL]
+        }
+
+    def get_filename(self) -> str:
+        return "digests/body"
+
+    def get_category(self) -> str:
+        return "digest_email"
+
+    def get_type(self) -> str:
+        return "notify.digest"
+
+    def get_unsubscribe_key(self) -> Optional[Tuple[str, int, Optional[str]]]:
+        return "project", self.project.id, "alert_digest"
+
+    def get_subject(self, context: Optional[Mapping[str, Any]] = None) -> str:
+        if not context:
+            # This shouldn't be possible but adding a message just in case.
+            return "Digest Report"
+        return get_digest_subject(context["group"], context["counts"], context["start"])
+
+    def get_notification_title(self) -> str:
+        # This shouldn't be possible but adding a message just in case.
+        return "Digest Report"
+
+    def get_reference(self) -> Any:
+        return self.project
+
+    def get_context(self) -> MutableMapping[str, Any]:
+        start, end, counts = get_digest_metadata(self.digest)
+        group = next(iter(counts))
+        return {
+            "counts": counts,
+            "digest": self.digest,
+            "end": end,
+            "group": group,
+            "has_alert_integration": has_alert_integration(self.project),
+            "project": self.project,
+            "slack_link": get_integration_link(self.organization, "slack"),
```

Comment:
what is `slack_link` for in this case? Is it only used later on in https://github.com/getsentry/sentry/pull/27969 ? I ask because as this PR is concerned we're not sending Slack notifications for digests, right?

Reply:
acutally nvm I see it in the old version too